### PR TITLE
dracut: Drop disable-xattr dracut patch, introduce config to optionally enable it

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Removes a hack to permanently disable copying of extended attributes for files in our initramfs. Instead introduce this behavior as an option by packaging DRACUT_NO_XATTR as a configuration in dracut-noxattr.rpm 

###### Change Log  

- Drop 0002-disable-xattr.patch
- Introduce dracut-noxattr subpackage to expose this behavior as an option

###### Does this affect the toolchain?  

NO

###### Associated issues  

- https://github.com/microsoft/azurelinux/issues/9580

###### Test Methodology

- Buddy build https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618543&view=results

### CODE REVIEW

**Comment by Camelron** [comment]
Depends on a separate change to build cp with xattr support: https://github.com/microsoft/azurelinux/commit/b6e4260797c6153ad16b9ebdb451d80ad0c8a671

**Comment by christopherco** [comment]
> Depends on a separate change to build cp with xattr support: [b6e4260](https://github.com/microsoft/azurelinux/commit/b6e4260797c6153ad16b9ebdb451d80ad0c8a671)

#10101

File: SPECS/dracut/50-noxattr.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# disable xattr
+DRACUT_NO_XATTR=1
```
Comment (christopherco) [inline SPECS/dracut/50-noxattr.conf:2]: Newline (please also update your editor to automatically include newlines in EOF)

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/dracut/50-noxattr.conf (hunk 1 above)
Comment (Camelron) [reply SPECS/dracut/50-noxattr.conf]: Done!

**Review by christopherco — APPROVED** [review]

**Pull request merged**